joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -866,10 +906,33 @@ controller_interface::CallbackReturn JointTrajectoryController::on_configure(
   resize_joint_trajectory_point(state_error_, dof_);
   resize_joint_trajectory_point(last_commanded_state_, dof_);
 
+  // create services
   query_state_srv_ = get_node()->create_service<control_msgs::srv::QueryTrajectoryState>(
     std::string(get_node()->get_name()) + "/query_state",
     std::bind(&JointTrajectoryController::query_state_service, this, _1, _2));
 
+  if (
+    !has_velocity_command_interface_ && !has_acceleration_command_interface_ &&
+    !has_effort_command_interface_)
+  {
+    scaling_factor_sub_ = get_node()->create_subscription<SpeedScalingMsg>(
+      "~/speed_scaling_input", rclcpp::SystemDefaultsQoS(),
```

Comment:
Wouldn't it make sense to make it transient local as we discussed before?

Reply:
My main concern of making the subscriber volatile is the incompatibility with a volatile publisher: https://docs.ros.org/en/rolling/Concepts/Intermediate/About-Quality-of-Service-Settings.html#qos-compatibilities

I agree, that having a volatile subscriber has its benefits. My concern is that this is not compatible with a `ros2 topic pub` call from the command line without any further parameters which might raise a lot of issues. Maybe we could try to mitigate that in the documentation?

Regarding the topic name: Yes, in my opinion this is a prime example use case for remappings.

Reply:
You are right about this. Nevertheless I think it is necessary to have it Transient local in order to avoid inconsistencies in the system if running multiple JTCs. Also there are plans to change this in the future: https://github.com/ros2/ros2/issues/1562

I agree regarding the topic name. The only argument I have for having the topic name in the parameter file is that it would allow to reconfigure the the topic at runtime (yes I sometimes do some wild stuff with controllers)... As far as I know this is not possible with remappings?!

Reply:
Let's go for the transient_local in combination with some documentation.

Regarding the topic: To be able to change it during runtime, there would be some other mechanisms necessary (such as restarting the subscriber in the parameter callback). I'm not sure whether such a use case should be modeled and I think I would like to prefer leaving that discussion out of this PR for now if that's fine with you. I see your point, but as you already stated: this might be kind of an esoteric use case ;-)